packages/api/lambdas/kinesis-consumer.js

```diff
@@ -75,25 +79,87 @@ function validateMessage(event) {
   return validate(event);
 }
 
+
+/**
+ * Publish a record to the fallback SNS topic for further processing.
+ *
+ * @param {Object} record - errored record
+ * @returns {Promise<Object>} - SNS publish response
+ */
+async function publishRecordToFallbackTopic(record) {
+  const fallbackArn = process.env.FallbackTopicArn;
+  log.info('publishing bad kinesis record to Topic:', fallbackArn);
+  log.info('record:', JSON.stringify(record));
+  return sns().publish({
+    TopicArn: fallbackArn,
+    Message: JSON.stringify(record)
+  }).promise();
+}
+
+
+/**
+ * processRecord error handler.  If the error comes on first attempt
+ *
+ * @param {Error} error - error raised in processRecord.
+ * @param {Object} record - record processed during error event.
+ * @param {Bool} shouldRetry - flag to determine if the error should be sent
+ *   for further processing or just raised to be handled by the
+ *   lambda. shouldRetry is true if the record being processes is directly from
+ *   Kinesis.
+ * @returns {(res|error)} - result of publishing to topic, or original error if publish fails.
+ * @throws {Error} - throws the original error if no special handling requested.
+ */
+function handleProcessRecordError(error, record, shouldRetry) {
+  if (shouldRetry) {
+    return publishRecordToFallbackTopic(record)
+      .then((res) => {
+        log.debug('sns result:', res);
+        return res;
+      })
+      .catch((snsError) => {
+        // TODO:  What should when we can't post to the fallback?
+        log.error(`Failed to publish record to fallback topic: ${record}`);
+        log.error(`original error: ${error}`);
+        log.error(`subsequent error: ${snsError}`);
+        return error;
+      });
+  }
+  throw error;
+}
+
 /**
  * Process data sent to a kinesis stream. Validate the data and
  * queue a workflow message for each rule.
  *
  * @param {*} record - input to the kinesis stream
+ * @param {Bool} fromSNS - flag specifying if this is event is from SNS.  SNS
+ *        events come the fallback notification and are re-tries of original
```

Comment:
I think this should be `SNS events come from the fallback SNS Topic and are retries...`